lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -38,10 +39,23 @@ class OneToManyPersister extends AbstractCollectionPersister
      */
     public function delete(PersistentCollection $collection)
     {
-        // This can never happen. One to many can only be inverse side.
-        // For owning side one to many, it is required to have a join table,
-        // then classifying it as a ManyToManyPersister.
-        return;
+        // The only valid case here is when you have weak entities. In this
+        // scenario, you have @OneToMany with orphanRemoval=true, and replacing
+        // the entire collection with a new would trigger this operation.
+        $mapping = $collection->getMapping();
+
+        if ( ! $mapping['orphanRemoval']) {
```

Comment:
Note that not doing that means also not going through the event listeners, and many folks rely on that as well